src/mono/browser/runtime/jiterpreter-trace-generator.ts

```diff
@@ -3977,8 +3977,12 @@ function emit_atomics (
     builder: WasmBuilder, ip: MintOpcodePtr, opcode: number
 ) {
     if (opcode === MintOpcode.MINT_MONO_MEMORY_BARRIER) {
-        if (WasmEnableThreads)
+        if (WasmEnableThreads) {
             builder.appendAtomic(WasmAtomicOpcode.atomic_fence);
+            // The text format and other parts of the spec say atomic.fence has no operands,
+            //  but the binary encoding requires a dummy zero byte for some reason
```

Comment:
It's for the memory ordering. See this part of the shared everythign threads proposal. 0 means sequential consistency, 1 will mean acquire-release https://github.com/WebAssembly/shared-everything-threads/blob/main/proposals/shared-everything-threads/Overview.md#memory-orderings

Reply:
> And sync_synchronize is sequential consistency, right?

It's a "full barrier" which presumably means seq cst. https://gcc.gnu.org/onlinedocs/gcc/_005f_005fsync-Builtins.html